server/server.c

```diff
@@ -120,503 +122,505 @@ jd_on_run (GThreadedSocketService* service, GSocketConnection* connection, GObje
+			gpointer iterator;
+			gconstpointer value;
+			guint32 len;
+			guint32 zero = 0;
 
-						key_len = strlen(key) + 1;
+			reply = j_message_new_reply(message);
+			namespace = j_message_get_string(message);
 
-						j_message_add_operation(reply, 4 + len + key_len);
-						j_message_append_4(reply, &len);
-						j_message_append_n(reply, value, len);
-						j_message_append_string(reply, key);
-					}
+			j_backend_kv_get_all(jd_kv_backend, namespace, &iterator);
 
-					j_message_add_operation(reply, 4);
-					j_message_append_4(reply, &zero);
+			while (j_backend_kv_iterate(jd_kv_backend, iterator, &key, &value, &len))
+			{
+				gsize key_len;
 
-					j_message_send(reply, connection);
-				}
-				break;
-			case J_MESSAGE_KV_GET_BY_PREFIX:
-				{
-					g_autoptr(JMessage) reply = NULL;
-					gchar const* prefix;
-					gpointer iterator;
-					gconstpointer value;
-					guint32 len;
-					guint32 zero = 0;
+				key_len = strlen(key) + 1;
 
-					reply = j_message_new_reply(message);
-					namespace = j_message_get_string(message);
-					prefix = j_message_get_string(message);
+				j_message_add_operation(reply, 4 + len + key_len);
+				j_message_append_4(reply, &len);
+				j_message_append_n(reply, value, len);
+				j_message_append_string(reply, key);
+			}
 
-					j_backend_kv_get_by_prefix(jd_kv_backend, namespace, prefix, &iterator);
+			j_message_add_operation(reply, 4);
+			j_message_append_4(reply, &zero);
 
-					while (j_backend_kv_iterate(jd_kv_backend, iterator, &key, &value, &len))
-					{
-						gsize key_len;
+			j_message_send(reply, connection);
+		}
+		break;
+		case J_MESSAGE_KV_GET_BY_PREFIX:
+		{
+			g_autoptr(JMessage) reply = NULL;
+			gchar const* prefix;
+			gpointer iterator;
+			gconstpointer value;
+			guint32 len;
+			guint32 zero = 0;
 
-						key_len = strlen(key) + 1;
+			reply = j_message_new_reply(message);
+			namespace = j_message_get_string(message);
+			prefix = j_message_get_string(message);
 
-						j_message_add_operation(reply, 4 + len + key_len);
-						j_message_append_4(reply, &len);
-						j_message_append_n(reply, value, len);
-						j_message_append_string(reply, key);
-					}
+			j_backend_kv_get_by_prefix(jd_kv_backend, namespace, prefix, &iterator);
 
-					j_message_add_operation(reply, 4);
-					j_message_append_4(reply, &zero);
+			while (j_backend_kv_iterate(jd_kv_backend, iterator, &key, &value, &len))
+			{
+				gsize key_len;
 
-					j_message_send(reply, connection);
-				}
-				break;
-			default:
+				key_len = strlen(key) + 1;
+
+				j_message_add_operation(reply, 4 + len + key_len);
+				j_message_append_4(reply, &len);
+				j_message_append_n(reply, value, len);
+				j_message_append_string(reply, key);
+			}
+
+			j_message_add_operation(reply, 4);
+			j_message_append_4(reply, &zero);
+
+			j_message_send(reply, connection);
+		}
+		break;
+		default:
+			if (!db_server_message_exec(message_type, message, operation_count, jd_db_backend, safety, connection))
```

Comment:
Why do you handle db messages in the default case?

Reply:
The function db_server_message_exec contains a switch case on the same variable, but with some additional code in front (and after) of the switch. It would be possible to define all the db-messages in cases with fall-through, and call the function there.